data/core/docview.lua

```diff
@@ -410,6 +410,10 @@ function DocView:draw()
   local minline, maxline = self:get_visible_line_range()
   local lh = self:get_line_height()
 
+  if self.size.y > 0 then
+    self.doc.highlighter:set_visible_lines(minline, maxline)
```

Comment:
One thing about this; in theory you can have several docviews to the same document (different splits working on the same file). 

As it stands, there are a number of issues with the way we handle this (like the fact that selection is at the doc level instead of the docview level), but this feature does work (and in future, we should probably move selection to docview, if we can do it without too much complications).

Given that, there is no single pair of visible line boundaries, but a list of them. I dunno if it's worth handling, because it's an edge case, but thought I'd point it out.

Reply:
Right, did not took this in consideration, will test, I may end simplifying this pr depending on the gains of #896